src/main/java/com/angelbroking/smartapi/http/SmartAPIRequestHandler.java

```diff
@@ -1,420 +1,379 @@
+    private static final String USER_AGENT = "javasmartapiconnect/3.0.0";
+    private final OkHttpClient client;
+    JSONObject apiheader = apiHeaders();
+
+    /**
+     * Initialize request handler.
+     *
+     * @param proxy to be set for making requests.
+     */
+    public SmartAPIRequestHandler(Proxy proxy) {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        builder.connectTimeout(10000, TimeUnit.MILLISECONDS);
+        if (proxy != null) {
+            builder.proxy(proxy);
+        }
+
+        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
+        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
+        if (SmartConnect.enableLogging) {
+            client = builder.addInterceptor(logging).build();
+        } else {
+            client = builder.build();
+        }
+    }
+
+    public JSONObject apiHeaders() {
+
+        try {
+            JSONObject headers = new JSONObject();
+
+            // Local IP Address
+            InetAddress localHost = InetAddress.getLocalHost();
+            String clientLocalIP = localHost.getHostAddress();
+            headers.put(Constants.HEADER_CLIENT_LOCAL_IP, clientLocalIP);
+
+            // Public IP Address
+            String clientPublicIP = getPublicIPAddress();
+            headers.put(Constants.HEADER_CLIENT_PUBLIC_IP, clientPublicIP);
+            String macAddress = null;
+            // MAC Address
+            // get all network interfaces of the current system
+            Enumeration<NetworkInterface> networkInterface = NetworkInterface.getNetworkInterfaces();
+            // iterate over all interfaces
+            while (networkInterface.hasMoreElements()) {
+                // get an interface
+                NetworkInterface network = networkInterface.nextElement();
+                // get its hardware or mac address
+                byte[] macAddressBytes = network.getHardwareAddress();
+                if (macAddressBytes != null) {
+                    // initialize a string builder to hold mac address
+                    StringBuilder macAddressStr = new StringBuilder();
+                    // iterate over the bytes of mac address
+                    for (int i = 0; i < macAddressBytes.length; i++) {
+                        // convert byte to string in hexadecimal form
+                        macAddressStr.append(String.format("%02X%s", macAddressBytes[i], (i < macAddressBytes.length - 1) ? "-" : ""));
+                    }
+
+                    macAddress = macAddressStr.toString();
+                    if (macAddress != null) {
+                        break;
+                    }
+                }
+            }
+            headers.put(Constants.MAC_ADDRESS, macAddress);
+            String accept = "application/json";
+            headers.put(Constants.ACCEPT, accept);
+            String userType = "USER";
+            headers.put(Constants.USER_TYPE, userType);
+            String sourceID = "WEB";

```

Comment:
Why declaring and adding it to header separately?

Reply:
The apiHeaders() method is generating a set of API headers for use in an API request. It is declaring and adding the headers separately in order to ensure that each header is properly set with the correct value.

Reply:
have removed the declaration, now we are only setting it to the class